opendevin/llm/llm.py

```diff
@@ -21,14 +22,16 @@ def __init__(self,
                  base_url=DEFAULT_BASE_URL,
                  num_retries=DEFAULT_LLM_NUM_RETRIES,
                  cooldown_time=DEFAULT_LLM_COOLDOWN_TIME,
+                 api_version=DEFAULT_API_VERSION,
```

Comment:
I'm not sure about setting this explicitly... Your script in the linked issue showed that the llm.completion() call worked without it, for Azure, didn't it? It may be because it's not necessary or because the user had already set it as env var, as we suggest [here](https://github.com/OpenDevin/OpenDevin/blob/main/docs/guides/AzureLLMs.md).

[LiteLLM documentation](https://docs.litellm.ai/docs/providers/azure) says it picks it from the env. IIRC its code is indeed picking it up from env if it's not set explicitly.

Otherwise, how did your script work? 😅

Reply:
That's why its name is AZURE_API_VERSION, not LLM_API_VERSION currently. It's not "our" var, it's set by the user to be picked up by LiteLLM.

I can see why this is a bit messy... Particularly for Azure since we support its vars for embeddings, but not for completion, which is reasonably perceived as inconsistent. The problem is, if we make a special case for Azure in completion, will we do the same for other cloud providers with multiple vars named their own ways? For [example](https://docs.litellm.ai/docs/providers/bedrock). So far we let _extra_ vars be in env, for this.

That said, Azure is clearly an on-going problem for users... WDYT @rbren about doing this... just for Azure?

Reply:
I think setting the `api_version` here should be fine--it will probably come in handy for other providers. Default is `None`, so this should be a no-op if it isn't set by the user